### PR TITLE
#5139 standard user session plugin

### PR DESCRIPTION
**Pull request by mbarto**
## Description

This work extends user sessions base support, to allow enabling them in the standard product with reasonable defaults, like storing the session on the browser localStorage.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#5139 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by tdipisa** [comment]
@offtherailz 

>moreover, should we enable user session in mapstore by default? What do you think? I think this is valid for georchestra (and it should be set enabled by default also in pluginsConfig.json for that project), but for MapStore, maybe, it should be disabled by default. @tdipisa, what do you think?

Generally speaking if we have it, we can test it. If we have it, we can use it and better maintain it over time: if this functionlity stays disabled we are never able to identify regressions at this stage.

**Comment by mbarto** [comment]
> Generally speaking if we have it, we can test it. If we have it, we can use it and better maintain it over time: if this functionlity stays disabled we are never able to identify regressions at this stage.
> The important thing is that in MS we use a client side session storage (and not DB).

I try to translate it in developer terms:
Generally speaking if we have it, we HAVE to test it. If we have it, we HAVE to use it and we have to MANTAIN it over time: if this functionlity stays disabled we are never able to identify regressions at this stage.

That said, no objections to have it enabled by default. Did you consider it also from the final user point of view? Does a user expect to see the last point of view when it reloads the map, instead of resetting to the standard view?

File: web/client/localConfig.json [hunk 1]
```diff
@@ -46,6 +46,9 @@
       {"name": "browser", "path": "browser"},
       {"name": "geostorymode", "path": "geostory.mode"},
       {"name": "featuregridmode", "path": "featuregrid.mode"}],
+    "userSessions": {
```
[the remark this answers is not in the document]
Comment (mbarto) [reply web/client/localConfig.json:49]: I don't have a strong opinion here. As a final user, I would expect to reset the map when hitting F5 on my browser.

**Comment by mbarto** [comment]
@offtherailz added the requested docs and pluginsConfig.json configuration for the UserSession plugin

**Comment by tdipisa** [comment]
@mbarto 
>That said, no objections to have it enabled by default. Did you consider it also from the final user point of view? Does a user expect to see the last point of view when it reloads the map, instead of resetting to the standard view?

I need this enabled by default in MS with a client side session storage to properly test it there too. It should be a matter of configuration, isn't it?

**Comment by mbarto** [comment]
@tdipisa I checked, and configuration enables user sessions on localStorage by default

**Comment by offtherailz** [comment]
I talked with @tdipisa : 
Is it possible to test this only in a context (and related maps) without setting it up globally, so we can avoid to configure session on master ? What do you think about this proposal @mbarto ?

**Comment by mbarto** [comment]
@offtherailz no, this configuration is not allowed at the moment: you mean to enable it only on one specific context or to enable it only for all the contexts?

File: web/client/localConfig.json [hunk 2]
```diff
@@ -46,6 +46,9 @@
...
+      "enabled": true
```
Comment (mbarto) [inline web/client/localConfig.json:50]:
```suggestion
      "enabled": false
```

**Comment by offtherailz** [comment]
At the end we decided with @tdipisa and @mbarto  to disable in MapStore, at least until the session cannot be disabled/enabled by the user. 
It will be enabled in GeOrchestra

**Pull request merged**